benchmarks/java/src/main/java/com/google/protocolbuffers/ProtoBenchCaliper.java

```diff
@@ -0,0 +1,210 @@
+      @Override
+      Message getDefaultInstance() {
+        return com.google.protobuf.benchmarks.BenchmarkMessage1Proto2.GoogleMessage1
+            .getDefaultInstance();
+      }
+      @Override String[] getSpecificCaliperOption() { return null; }
+    },
+    GOOGLE_MESSAGE2 {
+      @Override ExtensionRegistry getExtensionRegistry() { return ExtensionRegistry.newInstance(); }
+      @Override
+      Message getDefaultInstance() {
+        return com.google.protobuf.benchmarks.BenchmarkMessage2.GoogleMessage2.getDefaultInstance();
+      }
+      @Override String[] getSpecificCaliperOption() { return null; }
+    },
+    GOOGLE_MESSAGE3 {
+      @Override
+      ExtensionRegistry getExtensionRegistry() {
+        ExtensionRegistry extensions = ExtensionRegistry.newInstance();
+        benchmarks.google_message3.BenchmarkMessage38.registerAllExtensions(extensions);
+        benchmarks.google_message3.BenchmarkMessage37.registerAllExtensions(extensions);
+        benchmarks.google_message3.BenchmarkMessage36.registerAllExtensions(extensions);
+        benchmarks.google_message3.BenchmarkMessage35.registerAllExtensions(extensions);
+        benchmarks.google_message3.BenchmarkMessage34.registerAllExtensions(extensions);
+        benchmarks.google_message3.BenchmarkMessage33.registerAllExtensions(extensions);
+        benchmarks.google_message3.BenchmarkMessage32.registerAllExtensions(extensions);
+        benchmarks.google_message3.BenchmarkMessage31.registerAllExtensions(extensions);
+        benchmarks.google_message3.BenchmarkMessage3.registerAllExtensions(extensions);
+        return extensions;
+      }
+      @Override
+      Message getDefaultInstance() {
+        return benchmarks.google_message3.BenchmarkMessage3.GoogleMessage3.getDefaultInstance();
+      }
+      @Override 
+      String[] getSpecificCaliperOption() { 
+        String[] opt = new String[1];
+        opt[0] = "-Cinstrument.runtime.options.timingInterval=3000ms";
+        return opt;
+      }
+    },
+    GOOGLE_MESSAGE4 {
+      @Override
+      ExtensionRegistry getExtensionRegistry() {
+        ExtensionRegistry extensions = ExtensionRegistry.newInstance();
+        benchmarks.google_message4.BenchmarkMessage43.registerAllExtensions(extensions);
+        benchmarks.google_message4.BenchmarkMessage42.registerAllExtensions(extensions);
+        benchmarks.google_message4.BenchmarkMessage41.registerAllExtensions(extensions);
+        benchmarks.google_message4.BenchmarkMessage4.registerAllExtensions(extensions);
+        return extensions;
+      }
+      @Override
+      Message getDefaultInstance() {
+        return benchmarks.google_message4.BenchmarkMessage4.GoogleMessage4.getDefaultInstance();
+      }
+      @Override 
+      String[] getSpecificCaliperOption() { 
+        String[] opt = new String[1];
+        opt[0] = "-Cinstrument.runtime.options.timingInterval=3000ms";
```

Comment:
Can you just make it the default for everything and remove this method?

Reply:
Done, although this will make google_message1 and google_message2 somewhat slower than before